sensor/common/networkflow/manager/public_ips.go

```diff
@@ -58,21 +59,43 @@ func (m *publicIPsManager) Run(ctx concurrency.Waitable, clusterEntities EntityS
 	}
 }
 
+func (m *publicIPsManager) OnUpdate(ips set.Set[net.IPAddress]) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	for ip := range ips {
+		if _, ok := m.publicIPs[ip]; !ok {
+			// New public IP: trigger adding
+			m.onAddNoLock(ip)
+		}
+	}
+	for address := range m.publicIPs {
+		if !ips.Contains(address) {
+			// IP is tracked but not present in the update: trigger removal
+			m.onRemoveNoLock(address)
+		}
+	}
+}
+
 func (m *publicIPsManager) OnAdded(ip net.IPAddress) {
```

Comment:
Maybe remove it since it's not longer used.

Reply:
I didn't touch the unit tests for the `publicIPsManager` but I will remove it and update the tests.